packages/bug-detectors/index.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2023 Code Intelligence GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { registerCommandInjectionBugDetectors } from "./command-injection";
+
+export class BugDetectorError extends Error {}
+
+/**
+ * Registers bug detectors based on the provided list of bug detectors.
+ */
+export async function registerBugDetectors(
+	bugDetectors: string[]
+): Promise<void> {
+	// Keep track of registered bug detectors to avoid registering the same bug detector multiple times.
+	const registeredBugDetectors: string[] = [];
```

Comment:
We would need to extend this list every time we add a bug detector. Could we use the bug detector name to dynamically load it instead?

Reply:
It should just check itself if it's allowed to load itself. That should work, nice!